src/Interfaces/AuthenticationContainerInterface.php

```diff
@@ -0,0 +1,14 @@
+<?php
+declare(strict_types=1);
+
+namespace Firehed\API\Interfaces;
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * SHOULD use FQCN for get/has - etc
```

Comment:
Fill out a proper description here - but it's primarily for best practices.

Reply:
Note: the driving factor here is that authentication isn't always as simple as "do I have a user?", which is why this is so generic. Use cases:

- Provide time of authentication to the authorizer
- Provide method(s)/factor(s) of authentication to the authorizer
- Provide both the user and application from a bearer token
- Allow for weird admin panel type work, such as user impersonation